### PR TITLE
[FLOC-2422] Flocker deploy ip addresses

### PR DESCRIPTION
**Pull request by dwgebler**
Fixes FLOC-2422

### CODE REVIEW

**Comment by dwgebler** [comment]
I suppose another possibility is for the control service to attempt to resolve a hostname given in a deployment config to an IP (e.g. `socket.gethostbyname_ex`) and only return an error if it can't, but that's just a random thought, nothing to do with the scope of this issue.

**Comment by exarkun** [comment]
This also fixes #1366 1366

File: docs/using/config/configuration.rst [hunk 1]
```diff
@@ -300,9 +300,9 @@ Deployment Configuration
 ------------------------
 
 The deployment configuration specifies which applications are run on what nodes.
-It consists of a version and a mapping from node names to application names.
+It consists of a version and a mapping from node IP addresses to application names.
```
Comment (wallrj) [inline docs/using/config/configuration.rst]: "IPv4 address"

File: docs/using/config/configuration.rst [hunk 2]
```diff
@@ -311,7 +311,7 @@ Here's an example of a simple but complete configuration defining a deployment o
 
   "version": 1
   "nodes":
-    "node017.example.com":
+    "172.16.255.251":
```
Comment (wallrj) [inline docs/using/config/configuration.rst]: :question: Maybe use an IPv4 address reserved for documentation:
 * https://tools.ietf.org/html/rfc5737#section-3

File: flocker/control/_config.py [hunk 3]
```diff
@@ -1259,6 +1261,13 @@ def deployment_from_configuration(deployment_state, deployment_configuration,
     seen_applications = set()
     for hostname, application_names in (
             deployment_configuration['nodes'].items()):
+        try:
+            IPv4Address(hostname)
+        except AddressValueError:
+            raise ConfigurationError(
+                ("Node {node_name} specified, but deployment configuration "
+                 "expects an IPv4 address.").format(node_name=hostname)
+            )
```
Comment (wallrj) [inline flocker/control/_config.py]: :question: Maybe quote the incorrect hostname or it ends up looking like this:
```
(2422)[~/flocker-tutorial]$ flocker-deploy --certificates-directory=/tmp/tmpqgFZ7D 119.9.108.69 volume-deployment.yml volume-application.yml 
Node node2.example.com specified, but deployment configuration expects an IPv4 address.
```

**Comment by wallrj** [comment]
Thanks @dwgebler 

This seems to work.

And the code makes sense.

There's an acceptance test build failure, which I doubt is related to this branch, but there are no remote_logs so I can't tell:
 * http://build.clusterhq.com/builders/flocker%2Facceptance%2Faws%2Fubuntu-14.04%2Faws/builds/896/steps/trial/logs/remote_logs.log

 I think we're not currently merging anything unless all builds are green.

So you might want to force another build.

Merge if you can get green build results.

**Comment by wallrj** [comment]
Oh, and remember to close https://github.com/ClusterHQ/flocker/issues/1366 once you've merged this.

**Pull request merged**